libs/prelude/Decidable/Equality.idr

```diff
@@ -30,6 +22,20 @@ interface DecEq t where
   total decEq : (x1 : t) -> (x2 : t) -> Dec (x1 = x2)
 
 --------------------------------------------------------------------------------
+-- Utility lemmas
+--------------------------------------------------------------------------------
+
+||| The negation of equality is symmetric (follows from symmetry of equality)
+total negEqSym : {a : t} -> {b : t} -> (a = b -> Void) -> (b = a -> Void)
+negEqSym p h = p (sym h)
+
+||| Everything is decidably equal to itself
+total decEqSelfIsYes : DecEq a => {x : a} -> decEq x x = Yes Refl
+decEqSelfIsYes {x} with (decEq x x)
```

Comment:
Why should we have this in the prelude? Has it come up in practice for you? I imagine it's not so useful, because if I already know that something is the same as another, then I can just use `Refl` directly rather than calling `decEq` on it.